### PR TITLE
Make boot slot hooks more versatile

### PR DESCRIPTION
**Pull request by Bastian-Krause**
The i.MX28's ROM code expects an absolute offset to the raw device instead of an offset to a partition in the bootloader's header residing in a partition with a certain type. This is problematic when used in conjunction with `boot-mbr-switch`, because depending on which partition is active, the offset varies.

In order to allow such cases, run the pre install hook after the partition is determined and pass information required to manipulate the partition to the hook. Do the same for the post install hook, but run it prior to the actual atomic MBR switch.

To keep things consistent, apply these changes also to `boot-gpt-switch` and `boot-emmc`.

Please note that I only tested the `boot-mbr-switch` case, everything else is just compile-tested.

### CODE REVIEW

**Comment by Bastian-Krause** [comment]
Rebased on master.

File: src/update_handler.c [hunk 1]
```diff
@@ -748,14 +748,17 @@ static gboolean unpack_archive(RaucImage *image, gchar *dest, GError **error)
...
 {
 	g_autoptr(GSubprocessLauncher) launcher = NULL;
 	g_autoptr(GSubprocess) sproc = NULL;
 	g_autofree gchar* image_size = NULL;
 	GError *ierror = NULL;
 	gboolean res = FALSE;
 	RaucBundle *bundle;
+	GHashTableIter iter;
+	gchar *key = NULL;
+	gchar *value = NULL;
```
Comment (jluebbe) [inline src/update_handler.c:761]: This could be moved to the `if (variables)` block.

**Review by jluebbe — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: src/update_handler.c (hunk 1 above)
Comment (Bastian-Krause) [reply src/update_handler.c:761]: Good point, will do that.

**Comment by Bastian-Krause** [comment]
Force-pushed: integrated review feedback from @jluebbe, thanks!

**Review by jluebbe — APPROVED** [review]

**Pull request merged**